itest/rust/src/builtin_tests/script/script_instance_tests.rs

```diff
@@ -8,23 +8,36 @@
 use std::ffi::c_void;
 
 use godot::builtin::{Array, Dictionary, GString, StringName, Variant, VariantType};
-use godot::classes::{IScriptExtension, Object, Script, ScriptExtension, ScriptLanguage};
+use godot::classes::{
+    IScriptExtension, IScriptLanguageExtension, Object, Script, ScriptExtension, ScriptLanguage,
+    ScriptLanguageExtension,
+};
 use godot::global::{Error, MethodFlags};
 use godot::meta::{ClassName, FromGodot, MethodInfo, PropertyInfo, ToGodot};
 use godot::obj::script::{create_script_instance, ScriptInstance, SiMut};
-use godot::obj::{Base, Gd, WithBaseField};
+use godot::obj::{Base, Gd, NewAlloc, WithBaseField};
 use godot::register::{godot_api, GodotClass};
 use godot::sys;
 
+use crate::framework::itest;
+
 #[derive(GodotClass)]
-#[class(base = ScriptExtension, init, tool)]
+#[class(base = ScriptExtension, no_init, tool)]
 struct TestScript {
+    language: Gd<TestScriptLanguage>,
     base: Base<ScriptExtension>,
 }
 
+impl TestScript {
+    fn new(language: Gd<TestScriptLanguage>) -> Gd<Self> {
+        Gd::from_init_fn(|base| Self { language, base })
+    }
+}
```

Comment:
In practice, you'd probably have some sort of factory that hands out `TestScript` instances that refer to some internal `TestScriptLanguage` object, right?

(It's not clear how close the test is to "real-world code", maybe a brief comment would help...)

Reply:
I added a comment with real world implementation examples.